ShenkinStore/Controllers/ShoppingCartController.cs

```diff
@@ -0,0 +1,176 @@
+        {
+
+            var userID = HttpContext.Session.GetString("UserID");
+            if (userID != null)
+            {
+                var cart = ShoppingCart.GetCart(userID.ToString());
+
+                var viewModel = new ShoppingCartViewModel
+                {
+                    CartItems = cart.GetCartItems(),
+                    CartTotal = cart.GetTotal(),
+                    Amounts = cart.GetAmount(),
+                    TotalPricePerProduct = cart.GetProductTotal()
+                };
+                return View(viewModel);
+            }
+            else
+            {
+                return RedirectToAction("Index", "Home");
+            }
+        }
+
+
+        // GET: ShoppingCart/AddToCart/id
+        public ActionResult AddToCart(int id, int amount)
+        {
+            var userID = HttpContext.Session.GetString("UserID");
+            if (userID != null)
+            {
+                var addedProduct = db.Products.Single(product => product.ProductId == id);
+                var cart = ShoppingCart.GetCart(userID.ToString());
+
+
+                cart.AddToCart(addedProduct, amount);
+
+
+
+
+
+                return RedirectToAction("Index", "ShoppingCart");
+            }
+            else
+            {
+                return RedirectToAction("Login", "Users");
+            }
+        }
+
+
+        public ActionResult IncreaseProduct(int id)
+        {
+            var userID = HttpContext.Session.GetString("UserID");
+
+            var cart = ShoppingCart.GetCart(userID.ToString());
+            cart.IncreaseAmount(id);
+            return RedirectToAction("Index", "ShoppingCart");
+
+
+
+        }
+        public ActionResult DecreaseProduct(int id)
+        {
+            var userID = HttpContext.Session.GetString("UserID");
+
+            var cart = ShoppingCart.GetCart(userID.ToString());
+            cart.DecreaseAmount(id);
+            return RedirectToAction("Index", "ShoppingCart");
+
+
+        }
+
+        // GET: ShoppingCart/RemoveFromCart/id
+        public ActionResult RemoveFromCart(int id)
+        {
+            var userID = HttpContext.Session.GetString("UserID");
+            if (userID != null)
+            {
+                var cart = ShoppingCart.GetCart(userID.ToString());
+                Product product = db.Products.FirstOrDefault(item => item.ProductId == id);
+
+                cart.RemoveFromCart(product);
+                return RedirectToAction("Index", "ShoppingCart");
+            }
+            else
+            {
+                return RedirectToAction("Login", "Users");
+            }
+
+        }
+
+        public ActionResult emptyCart()
+        {
+            var userID = HttpContext.Session.GetString("UserID");
+            if (userID != null)
+            {
+
+                var cart = ShoppingCart.GetCart(userID.ToString());
+                var cartItems = cart.GetCartItems();
+                foreach (var cartItem in cartItems)
```

Comment:
I think Maps have RevmoveAll built in so add a function in cart that does remove all and replace with this foreach